core/macros/src/class.rs

```diff
@@ -0,0 +1,824 @@
+                }
+            })
+            .collect::<SpannedResult<_>>()?;
+
+        let length = take_length_from_attrs(&mut fn_.attrs)?.unwrap_or(args_decl.len());
+        let fn_name = &fn_.sig.ident;
+
+        // Does the function return Result<_> or JsResult<_>? If so, Into JsResult (to
+        // transform the error. If not, return Ok(_).
+        let return_statement = match &fn_.sig.output {
+            ReturnType::Default => quote! { Default::default() },
+            ReturnType::Type(_, ty) => {
+                if let Type::Path(path) = ty.as_ref() {
+                    let Some(t) = path.path.segments.last() else {
+                        return error(&fn_.sig.output, "Cannot infer return type.");
+                    };
+                    if t.ident == "Self" {
+                        quote! { Ok(result) }
+                    } else if t.ident == "JsResult" {
+                        quote! { result.into() }
+                    } else {
+                        return error(&fn_.sig.output, "Invalid return type.");
+                    }
+                } else {
+                    quote! { Ok(result) }
+                }
+            }
+        };
+
+        Ok(Self {
+            length,
+            name: String::new(),
+            body: quote! {
+                let rest = args;
+                #(#args_decl)*
+
+                let result = Self:: #fn_name ( #(#args_call),* );
+                #return_statement
+            },
+            is_static: false,
+        })
+    }
+}
+
+#[derive(Debug, Default)]
+struct Accessor {
+    getter: Option<Function>,
+    setter: Option<Function>,
+}
+
+impl Accessor {
+    fn set_getter(
+        &mut self,
+        name: String,
+        fn_: &mut ImplItemFn,
+        class_ty: &Type,
+    ) -> SpannedResult<()> {
+        let getter = Function::getter(name, fn_, class_ty)?;
+        if self.getter.is_some() {
+            error(fn_, "Getter for this property already declared.")
+        } else {
+            self.getter = Some(getter);
+            Ok(())
+        }
+    }
+
+    fn set_setter(
+        &mut self,
+        name: String,
+        fn_: &mut ImplItemFn,
+        class_ty: &Type,
+    ) -> SpannedResult<()> {
+        let setter = Function::setter(name, fn_, class_ty)?;
+        if self.setter.is_some() {
+            error(fn_, "Setter for this property already declared.")
+        } else {
+            self.setter = Some(setter);
+            Ok(())
+        }
+    }
+
+    fn body(&self) -> TokenStream2 {
+        let Some(name) = self
+            .getter
+            .as_ref()
+            .map_or_else(|| self.setter.as_ref().map(|s| &s.name), |g| Some(&g.name))
+        else {
+            return quote! {};
+        };
+        let getter = if let Some(getter) = self.getter.as_ref() {
+            let body = getter.body.clone();
+            quote! {
+                Some(
+                    boa_engine::NativeFunction::from_copy_closure( #body )
```

Comment:
From what I can tell (after replacing it locally running the tests), we can use `NativeFunction::from_fn_ptr()` instead of closure, which is faster and smaller in size. (should replace all instances)

Reply:
Let me check, that might be nice.

Reply:
Just to note, all instances of `from_copy_closure` should be replaced :smile:

Reply:
Uh, you're right. Dunno why I just replaced the one. Done.